File: Examples/iOSDemo/CanaryDemo/ViewController.m
//
//  ViewController.m
//  CanaryDemo
//
//  Created by Rake Yang on 2020/12/13.
//

#import "ViewController.h"
#import <AFNetworking/AFNetworking.h>
#import "CanaryDemo-Bridging-Header.h"
#import "CanaryDemo-Swift.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UITextView *logView;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    [self fetchParam];
}

- (void)fetchParam {
    [self showLog:[NSString stringWithFormat:@"获取配置参数：A = %@", [CanaryManager.shared stringValueFor:@"A" def:@"123"]]];
}

- (void)showLog:(NSString *)log {
    NSLog(@"%@", log);
    dispatch_async(dispatch_get_main_queue(), ^{
        self.logView.text = log;
    });
}

- (IBAction)showCanary:(id)sender {
    [CanaryManager.shared show];
    [self fetchParam];
    DDLogVerbose(@"verbose");
    DDLogInfo(@"info");
    DDLogWarn(@"warn");
    DDLogDebug(@"degbu");
    DDLogError(@"error");
}

- (IBAction)showNetworking:(id)sender {
    
    NSLog(@"%@", NSURLSessionConfiguration.defaultSessionConfiguration.protocolClasses);
    
    AFHTTPSessionManager *manager = [[AFHTTPSessionManager alloc] initWithSessionConfiguration:NSURLSessionConfiguration.defaultSessionConfiguration];
    manager.requestSerializer = AFHTTPRequestSerializer.serializer;
    manager.responseSerializer = AFJSONResponseSerializer.serializer;
    NSURLSessionDataTask *task = [manager GET:@"https://api.m.taobao.com/rest/api3.do?api=mtop.common.getTimestamp" parameters:nil headers:nil progress:nil success:^(NSURLSessionDataTask * _Nonnull task, id  _Nullable responseObject) {
        [self showLog:[NSString stringWithFormat:@"%@", responseObject]];
    } failure:^(NSURLSessionDataTask * _Nullable task, NSError * _Nonnull error) {
        [self showLog:[NSString stringWithFormat:@"%@", error]];
    }];
    [task resume];
    
}

- (IBAction)showNetworkingParam:(id)sender {
    NSLog(@"%@", NSURLSessionConfiguration.defaultSessionConfiguration.protocolClasses);
    
    AFHTTPSessionManager *manager = [[AFHTTPSessionManager alloc] initWithSessionConfiguration:NSURLSessionConfiguration.defaultSessionConfiguration];
    manager.requestSerializer = AFHTTPRequestSerializer.serializer;
    manager.responseSerializer = AFJSONResponseSerializer.serializer;
    NSURLSessionDataTask *task = [manager GET:@"http://quan.suning.com/getSysTime.do" parameters:nil headers:nil progress:nil success:^(NSURLSessionDataTask * _Nonnull task, id  _Nullable responseObject) {
        [self showLog:[NSString stringWithFormat:@"%@", responseObject]];
    } failure:^(NSURLSessionDataTask * _Nullable task, NSError * _Nonnull error) {
        [self showLog:[NSString stringWithFormat:@"%@", error]];
    }];
    [task resume];
}


@end

File: iOS-Example/Canary/ViewController.swift
//
//  ViewController.swift
//  Canary
//
//  Created by Rake Yang on 03/18/2020.
//  Copyright (c) 2020 Rake Yang. All rights reserved.
//

import UIKit
import Canary

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    @IBAction func showConfig(_ sender: Any) {
        DDLogVerbose("verbose...")
        DDLogDebug("debug...")
        DDLogInfo("info...")
        DDLogWarn("warning...")
        DDLogError("error")
        
        CNManager()?.show()
    }
    
    @IBAction func showNetLog(_ sender: Any) {
        let request = URLRequest.init(url: URL.init(string: "http://127.0.0.1:8081/api/canary/conf")!)
        URLSession.shared.dataTask(with: request) { (data, response, error) in
            print("\(String(data: data ?? Data(), encoding: .utf8))")
        }.resume()
    }
    
    @IBAction func showMockData(_ sender: Any) {
        CanarySwift.shared.showMock()
    }
    
    @IBAction func showDoraemon(_ sender: Any) {
    }

}
